docs/sources/datasources/elasticsearch/configure-elasticsearch-data-source.md

```diff
@@ -0,0 +1,153 @@
+
+- **Pattern** - Select the matching pattern if using one in your index name. Options include:
+
+  - no pattern
+  - hourly
+  - daily
+  - weekly
+  - monthly
+  - yearly
+
+- **Time field name** - Name of the time field. The default value is @timestamp. You can enter a different name.
+
+- **Max concurrent shard requests** - Sets the number of shards being queried at the same time. The default is `5``. For more information on shards see [Elasticsearch's documentation](https://www.elastic.co/guide/en/elasticsearch/reference/8.9/scalability.html#scalability).
+
+- **Min time interval** - Defines a lower limit for the auto group-by time interval. This value **must** be formatted as a number followed by a valid time identifier:
+
+  | Identifier | Description |
+  | ---------- | ----------- |
+  | `y`        | year        |
+  | `M`        | month       |
+  | `w`        | week        |
+  | `d`        | day         |
+  | `h`        | hour        |
+  | `m`        | minute      |
+  | `s`        | second      |
+  | `ms`       | millisecond |
+
+We recommend setting this value to match your Elasticsearch write frequency.
+For example, set this to `1m` if Elasticsearch writes data every minute.
+
+You can also override this setting in a dashboard panel under its data source options. The default is `10s`.
+
+- **X-Pack enabled** - Toggle to enable `X-Pack`-specific features and options, which provide the [query editor]({{< relref "./query-editor/" >}}) with additional aggregations, such as `Rate` and `Top Metrics`.
+
+- **Include frozen indices** - Toggle on when the `X-Pack enabled` setting is active and the configured Elasticsearch version is higher than `6.6.0`. You can configure Grafana to include [frozen indices](https://www.elastic.co/guide/en/elasticsearch/reference/7.13/frozen-indices.html) when performing search requests.
+
+{{% admonition type="note" %}}
+Frozen indices are [deprecated in Elasticsearch](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/frozen-indices.html) since v7.14.
+{{% /admonition %}}
+
+### Logs
+
+In this section you can configure which fields the data source uses for log messages and log levels.
+
+- **Message field name:** - Grabs the actual log message from the default source.
+
+- **Level field name:** - Name of the field that has log level/severity information. When a level label is specified, the value of this label is used to determine the log level and update color of each log line accordingly. If the log doesn’t have specified level label, we try to find out if its content matches any of the [supported expressions](/docs/grafana/latest/explore/logs-integration/#log-level). The log level is always determined by the first match. In the case where Grafana is not able to infer a log level field, it will be visualized with an unknown log level.
```

Comment:
```suggestion
- **Level field name:** - Name of the field with log level/severity information. When a level label is specified, the value of this label is used to determine the log level and update the color of each log line accordingly. If the log doesn’t have a specified level label, we try to determine if its content matches any of the [supported expressions](/docs/grafana/latest/explore/logs-integration/#log-level). The first match always determines the log level. If Grafana cannot infer a log-level field, it will be visualized with an unknown log level.
```